### ISSUE TITLE
unexpected gradient() behaviour

### ISSUE THREAD

**Issue by manuel-delverme** [google/jax#4730]
https://github.com/google/jax/blob/dad8377b78522353358da75aff7a040a5cfc2fae/jax/api.py#L786:L791

right now jax.grad behaves like this:
`gx, = grad(f, (0,))(x0)`
`gx, gy = grad(f, (0,1))(x0, y0)`

A more intuitive behavior would be:
`gx = grad(f, (0,))(x0)`
`gx, gy = grad(f, (0,1))(x0, y0)`

The extra comma is easily missed because the gradient w.r.t. one variable is a variable, would you be interested in a PR changing the grad() behaviour to the second case?

**Comment by mattjj** [comment]
Hey, thanks for the suggestion!

I think no, we want to keep the behavior as-is, for a couple reasons:
1. You can write `gx = grad(f, 0)(x0)` now, so there's already a way to express what you want.
2. The rule right now is pretty simple: if you pass in `argnums` as a tuple, you get out a tuple result, and if you pass it in as an int, you get out a result corresponding to a single argument. That is, the behavior depends on the type (int vs tuple) of the value passed in as `argnums`. AIUI the alternative rule would make the behavior depend on the type _and the length_ of the value passed to `argnums`.

WDYT?

**Comment by manuel-delverme** [comment]
Makes sense, thanks.
